crypto/nextgen_crypto/src/ed25519.rs

```diff
@@ -481,6 +481,20 @@ pub mod compat {
         }
     }
 
+    /// Used to produce keypairs from a seed for testing purposes
+    #[cfg(any(test, feature = "testing"))]
+    pub fn keypair_strategy() -> impl Strategy<Value = (Ed25519PrivateKey, Ed25519PublicKey)> {
+        // The no_shrink is because keypairs should be fixed -- shrinking would cause a different
+        // keypair to be generated, which appears to not be very useful.
+        any::<[u8; 32]>()
+            .prop_map(|seed| {
+                let mut rng: StdRng = SeedableRng::from_seed(seed);
+                let (private_key, public_key) = generate_keypair(&mut rng);
```

Comment:
Just realized that `generate_keypair` is not annotated for `testing`. Let's take the opportunity and update this as well.

Reply:
I would, but that annotation would be a PITA to propagate through the large amount of proptest code in `types` without https://github.com/libra/libra/pull/356

Actually, propagating this is pretty much exactly #356